apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/TrimService.java

```diff
@@ -60,6 +85,11 @@ public void trimDerivedTables(int height) {
             doTrimDerivedTablesOnBlockchainHeight(height);
             log.debug("Total trim time: " + (System.currentTimeMillis() - startTime));
             dataSource.commit(!inTransaction);
+            if (async) {
+                trimEvent.select(new AnnotationLiteral<Async>() {}).fire(getLastTrimHeight());
```

Comment:
Is it fireAsync() method instead of fire() ??

Reply:
No we use such approach, because shard process is not fully asynchronous and we require fully synchronous sharding when restoring trim operations